### PR TITLE
#24 Issue - pokemon_type and (now removed) pokemon_weakness

### PR DESCRIPTION
**Pull request by amywhying**
- Added the following methods in `/pokemon/views.py` related to pokemon_type:
    - `find_all_pokemon_types`
    - `find_all_pokemon_types_by_pokemon_id`
    - `find_pokemon_type_by_pokemon_id_and_type`
    - `create_pokemon_type`
    - `update_pokemon_type`
    - `delete_pokemon_type`

- Modified `populate_test_data.py` to generate data for pokemon_type
- Removed pokemon_weakness model because of no API to get its data

### CODE REVIEW

File: pokedex/pokemon/views.py [hunk 1]
```diff
@@ -99,6 +100,57 @@ def delete_pokemon_info(request: HttpRequest, national_num: int) -> HttpResponse
...
+    
+def find_all_pokemon_types_by_pokemon_id(_: HttpRequest, national_num: int) -> HttpResponse:
+    pokemon_types = PokemonType.objects.all(pokemon_info__pk=national_num)
+    return HttpResponse(to_json(pokemon_types))
+
+def find_pokemon_type_by_pokemon_id_and_type(_: HttpRequest, national_num: int, type: str) -> HttpResponse:
+    pokemon_type = get_object_or_404(PokemonType, type=type, pokemon_info__pk=national_num)
+    return HttpResponse(to_json_one(pokemon_type))
+    
+# The body request should include a national_num identifying which pokemoin_info we are looking at
```
Comment (CookieComputing) [inline pokedex/pokemon/views.py:116]: nit: pokemon_info

**Review by CookieComputing — APPROVED** [review]
lgtm

File: pokedex/pokemon/views.py (hunk 1 above)
Comment (amywhying) [reply pokedex/pokemon/views.py:116]: nit?

Comment (CookieComputing) [reply pokedex/pokemon/views.py:116]: nitpick- small thing that doesn't affect the PR quality but is just a quick fix

Comment (amywhying) [reply pokedex/pokemon/views.py:116]: ooo what do u want me to change it to?

Comment (CookieComputing) [reply pokedex/pokemon/views.py:116]: it's a typo haha: `pokemoin_info` to `pokemon_info`

Comment (amywhying) [reply pokedex/pokemon/views.py:116]: OMGGGG 
okok let me change

**Pull request merged**